sections/shorting/ShortingRewards/ShortingRewardRow.tsx

```diff
@@ -0,0 +1,214 @@
+
+import { CRYPTO_CURRENCY_MAP } from 'constants/currency';
+import { CurrencyKey } from 'constants/currency';
+
+import useCollateralShortRewards from 'queries/collateral/useCollateralShortRewards';
+import Card from 'components/Card';
+import Currency from 'components/Currency';
+
+type ShortingRewardRowProps = {
+	currencyKey: CurrencyKey;
+	gasPrice: number | null;
+	setGasLimit: Dispatch<SetStateAction<number | null>>;
+	snxPriceRate: number;
+};
+
+const ShortingRewardRow: FC<ShortingRewardRowProps> = ({
+	currencyKey,
+	gasPrice,
+	setGasLimit,
+	snxPriceRate,
+}) => {
+	const { t } = useTranslation();
+
+	const { notify } = Connector.useContainer();
+	const { monitorHash } = Notify.useContainer();
+
+	const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+	const [txError, setTxError] = useState<string | null>(null);
+	const walletAddress = useRecoilValue(walletAddressState);
+	const [txConfirmationModalOpen, setTxConfirmationModalOpen] = useState<boolean>(false);
+
+	const { selectPriceCurrencyRate } = useSelectedPriceCurrency();
+	const collateralShortRewardsQuery = useCollateralShortRewards(currencyKey);
+
+	const ShortingRewardRow = useMemo(
+		() =>
+			collateralShortRewardsQuery.isSuccess ? collateralShortRewardsQuery?.data ?? null : null,
+		[collateralShortRewardsQuery.isSuccess, collateralShortRewardsQuery.data]
+	);
+
+	const submissionDisabledReason = useMemo(() => {
+		if (isSubmitting) {
+			return t('shorting.rewards.button.claiming');
+		}
+		if (ShortingRewardRow == null || ShortingRewardRow.lte(0)) {
+			return t('shorting.rewards.button.claim');
+		}
+		return null;
+	}, [ShortingRewardRow, isSubmitting, t]);
+
+	const isSubmissionDisabled = useMemo(() => (submissionDisabledReason != null ? true : false), [
+		submissionDisabledReason,
+	]);
+
+	const totalTradePrice = useMemo(() => {
+		if (ShortingRewardRow != null) {
+			let tradePrice = ShortingRewardRow.multipliedBy(snxPriceRate);
+			if (selectPriceCurrencyRate) {
+				tradePrice = tradePrice.dividedBy(selectPriceCurrencyRate);
+			}
+
+			return tradePrice;
+		}
+		return 0;
+	}, [ShortingRewardRow, snxPriceRate, selectPriceCurrencyRate]);
+
+	const getGasEstimate = useCallback(async () => {
+		if (synthetix.js != null && walletAddress != null) {
+			try {
+				const { CollateralShort } = synthetix.js.contracts;
+
+				const gasLimitEstimate = await CollateralShort.estimateGas.getReward(
+					ethers.utils.formatBytes32String(currencyKey),
+					walletAddress
+				);
+
+				return normalizeGasLimit(Number(gasLimitEstimate));
+			} catch (e) {
+				console.log('gas limit error:', e);
+				return null;
+			}
+		}
+		return null;
+	}, [walletAddress, currencyKey]);
+
+	useEffect(() => {
+		async function getGasEstimateCall() {
+			const newGasLimit = await getGasEstimate();
+			setGasLimit(newGasLimit);
+		}
+		getGasEstimateCall();
+	}, [getGasEstimate, setGasLimit]);
+
+	const onSubmit = async () => {
```

Comment:
wrap this in a useCallback(async () => { }, [...props]) since we are passing into the Button component